tests/system_test.py

```diff
@@ -372,6 +372,10 @@ def error(msg):
         # at this point the process has terminated, either of its own accord or
         # due to the above terminate call.
 
+        # prevent leak of stdin fd
```

Comment:
Is closing the stdin the only fix needed in order not to see the backtrace seen in https://github.com/skupperproject/skupper-router/issues/415 ?

Reply:
Yes, close stdin and ensure that the `teardown` method is always called.

Reply:
Ok, so you added the `__enter__` and `__exit__` functions to `class Qdrouterd` so if anyone uses the `with` keyword, we can ensure that the `teardown()` function will be called. But adding these two functions and the changes to the other tests in this commit are not part of the fix. The *actual* fix is to close the stdin on `teardown()`, right ?

Reply:
It is part of the fix. I need to ensure that stain gets closed. If some tests never called teardown() and I am closing stdin in teardown, then I have to also call teardown there.

Am I missing something obvious here?

Reply:
I think I get it now. The change in system_tests_management.py from assertTrue to assertEqual is unrelated, yes. Let me put it to separate PR.

Reply:
Split into https://github.com/skupperproject/skupper-router/pull/508, thanks!